sandpack-react/src/common/BareCodeSandboxButton.tsx

```diff
@@ -0,0 +1,96 @@
+import type { SandpackBundlerFiles } from "@codesandbox/sandpack-client";
+import { getParameters } from "codesandbox-import-utils/lib/api/define";
+import * as React from "react";
+
+import { useSandpack } from "../hooks/useSandpack";
+import type { SandboxEnvironment } from "../types";
+
+const CSB_URL = "https://codesandbox.io/api/v1/sandboxes/define";
+
+const getFileParameters = (
+  files: SandpackBundlerFiles,
+  environment?: SandboxEnvironment
+) => {
+  type NormalizedFiles = Record<
+    string,
+    {
+      content: string;
+      isBinary: boolean;
+    }
+  >;
+
+  const normalizedFiles = Object.keys(files).reduce((prev, next) => {
+    const fileName = next.replace("/", "");
+    const value = {
+      content: files[next].code,
+      isBinary: false,
+    };
+
+    return { ...prev, [fileName]: value };
+  }, {} as NormalizedFiles);
+
+  return getParameters({
+    files: normalizedFiles,
+    ...(environment ? { template: environment } : null),
+  });
+};
+
+export const BareCodeSandboxButton: React.FC<{ className?: string }> = ({
+  children,
+  ...props
+}) => {
+  const [paramsValues, setParamsValues] = React.useState("");
+  const { sandpack } = useSandpack();
+  const formRef = React.useRef<HTMLFormElement>(null);
+
+  React.useEffect(
+    function debounce() {
+      const timer = setTimeout(() => {
+        const params = getFileParameters(sandpack.files, sandpack.environment);
+
+        setParamsValues(params);
+      }, 600);
+
+      return () => {
+        clearTimeout(timer);
+      };
+    },
+    [sandpack.environment, sandpack.files]
+  );
+
+  // Register the usage of the codesandbox link
+  React.useEffect(function registerUsage() {
+    sandpack.openInCSBRegisteredRef.current = true;
+  }, []);
+
+  /**
+   * This is a safe limit to avoid too long requests (401),
+   * as all parameters are attached in the URL
+   */
+  if (paramsValues.length > 1500) {
```

Comment:
What do you think about this conditional? Doesn't it sound arbitrary?

Reply:
limit is 2000 for the entire URL as far as I know, I think it's a fair value, as there are several other elements of the URL maybe extract it in a constant somewhere? with an explanation (brief) about the value